### PR TITLE
92 ui validations

### LINKED ISSUE #92: Missing needed info: As a provider who has uploaded data, I can see when required fields are missing.
## On each field required for data collection:
Screen Shot 2020-08-31 at 3 05 25 PM
classes in ECE Reporter: 

```
<span class="usa-warning-message" id="birthCertificateId-birthState-birthTown-warning" role="status"><span class="oec-inline-icon oec-inline-icon--incomplete">[SVG goes here]<span class="usa-sr-only">(incomplete)</span></span> This information is required for OEC reporting</span>

<div class=" usa-form-group usa-form-group--warning">
<label class="usa-label usa-label--warning" for="birthCertificateId">Birth certificate ID # </label>
<input class="usa-input usa-input--warning" type="text" id="birthCertificateId" aria-describedby="birthCertificateId-warning" aria-invalid="false" aria-required="true" value="">
</div>

```

Screen Shot 2020-08-31 at 3 12 30 PM

```

<span class="oec-inline-icon oec-inline-icon--incomplete">[SVG goes here]<span class="usa-sr-only">(incomplete)</span></span>


```

## On each child roster item with missing/needed info:
Screen Shot 2020-08-31 at 3 06 54 PM

```
<span class="oec-inline-icon oec-inline-icon--incomplete">[SVG goes here]<span class="usa-sr-only">(incomplete)</span></span>

```

## On each child record section:
Screen Shot 2020-08-31 at 3 09 37 PM

## Alert on top of roster:
Update roster

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #92

### CODE REVIEW

**Comment by mmazanec22** [comment]
Still need to block errors from showing on first load and set the focus to the first error.  Sorry for the obnoxious number of files changed-- I changed the EditForms directory to just Forms because most of them are reused when creating a child.

File: client/public/index.html [hunk 1]
```diff
@@ -11,7 +11,11 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico" />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-    <link rel="mask-icon" href="%PUBLIC_URL%/safari-pinned-tab.svg" color="#61366f">
+    <link
```
Comment (mmazanec22) [inline client/public/index.html:14]: Runaway manual prettier

File: client/src/components/Forms/EnrollmentFunding/Fields/Site.tsx [hunk 2]
```diff
@@ -9,6 +9,10 @@ import {
 } from '@ctoec/component-library';
 import { Site, Enrollment } from '../../../../shared/models';
 import { ChangeEnrollment } from '../../../../shared/payloads';
+import {
+  getValidationStatusForField,
```
Comment (jlhogan) [inline client/src/components/Forms/EnrollmentFunding/Fields/Site.tsx:13]: unused imports?

File: client/src/utils/getValidationStatus.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+>(
+  objectDriller: ObjectDriller<T>,
+  path: string,
+  options?: ValidationStatusOptions
+): FormStatusProps | undefined {
+  const splitPath = path.split('.');
+  const field = splitPath.pop();
+
+  let _drilledObject: any = objectDriller;
+  if (splitPath.length) {
```
Comment (jlhogan) [inline client/src/utils/getValidationStatus.ts:22]: I think making the change i suggested in the component lib saves the util from needing to this. Should be able to just pass in an instance of objectDriller that as already navigated to prop

File: client/src/utils/getValidationStatus.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  return {
+    type: 'warning',
+    id: `status-${fields.join('-')}`,
+    message,
+    ...options,
+  };
+}
+
+export function distributeValidationErrorsToSubObjects(
```
Comment (jlhogan) [inline client/src/utils/getValidationStatus.ts:67]: can you add a comment explaining why this is needed?

File: src/entity/decorators/momentValidators.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { registerDecorator, ValidationOptions } from 'class-validator';
+import { Moment } from 'moment';
+
+const defaultMomentMessage = 'Date is invalid';
+
+export function MomentComparison(
```
Comment (jlhogan) [inline src/entity/decorators/momentValidators.ts:6]: also comments here please!

**Review by jlhogan — APPROVED** [review]
overall looks good barring the suggestion in comment/ component lib

File: client/src/utils/getValidationStatus.ts (hunk 3 above)
Comment (mmazanec22) [reply client/src/utils/getValidationStatus.ts:22]: We need to get the parent object though, to see if it has validation errors-- already navigated objectDriller value is just the value of the field

Comment (jlhogan) [reply client/src/utils/getValidationStatus.ts:22]: duhhh!!! ok v smart, component lib PR approved !!

File: client/src/components/Forms/ChildInfo/Fields/Race.tsx [hunk 6]
```diff
@@ -8,6 +8,34 @@ import {
   Checkbox,
 } from '@ctoec/component-library';
 import { Child } from '../../../../shared/models';
+import { getValidationStatusForFieldset } from '../../../../utils/getValidationStatus';
+
+/**
+ * Helper type of all valid race properties on Child
+ */
+type RaceField =
```
Comment (JordanGuinn) [inline client/src/components/Forms/ChildInfo/Fields/Race.tsx:16]: nit: Thoughts on using `Extract` here so we be sure that these fields and the attributes on `Child` are always equivalent?  Something like:

```
type RaceField = Extract<keyof Child, 
  | 'americanIndianOrAlaskaNative'
  | 'asian'
  | 'blackOrAfricanAmerican'
  | 'nativeHawaiianOrPacificIslander'
  | 'white'
>;
```

That way we can ensure type safety between `RaceField` and `Child` and avoid any potential spelling issues down the road.

**Review by JordanGuinn — APPROVED** [review]
LGTM - just a minor nitpick 👍🏽👍🏽

**Pull request merged**